src/pam_google_authenticator.c

```diff
@@ -1890,6 +1873,24 @@ static int google_authenticator(pam_handle_t *pamh,
       log_message(LOG_WARNING , pamh, "No secret configured for user %s, asking for code anyway.", username);
     }
 
+#ifdef ENABLE_AUTHY
+    if (params.enable_authy) {
+      authy_rc_t arc;
+      long authy_id = get_cfg_value_long(pamh, buf, "AUTHY_ID");
+      long authy_timeout = get_cfg_value_long(pamh, buf, "AUTHY_TIMEOUT");
+      if (authy_timeout < 0) {
+        authy_timeout = 30;
+      }
+      char *api_key = (char *)get_cfg_value_char(pamh, buf, "AUTHY_API_KEY");
+      arc = authy_login(pamh, authy_id, api_key, authy_timeout);
```

Comment:
Have as much code as possible be not-ifdefed.

How about this: create a second file called `pam_no_authy.c`  with `an authy_login()` implementation that always fails, logging that module was not built with authy support. That way no added `ifdef` in code, only in the Makefile.

Reply:
I wanted to minimize the changes introduced to the original code.

I see your point, I'll create a dummy `authy_login()` function.